src/modules/loops/iter_loop.rs

```diff
@@ -143,6 +152,114 @@ impl IterLoop {
+
+        // Define index handling if needed
+        let (index_init, index_update) = match (self.iter_index.as_ref(), self.iter_index_global_id) {
+            (Some(index), Some(global_id)) => {
+                let idx_var = get_variable_name(index, Some(global_id));
+                (raw_fragment!(", {idx_var}=0"), raw_fragment!(", {idx_var}++"))
+            },
+            _ => (FragmentKind::Empty, FragmentKind::Empty)
+        };
+
+        let body = self.block.translate(meta);
+
+        let init = fragments!(iter_name.clone(), "=", from_var, index_init);
+        // We do a trick here by multiplying by dir so that we can use the same comparison operator
+        // iter_name * dir_var < to_var * dir_var
+        let cond = fragments!(iter_name.clone(), " * ", dir_var.clone(), " ", op, " ", to_var, " * ", dir_var.clone());
+        let update = fragments!(iter_name, "+=", dir_var, index_update);
+
+        let stmt = fragments!(
+            "for (( ", init, "; ", cond, "; ", update, " )); do\n",
+            body,
+            "\ndone"
+        );
+        
+        stmt
+    }
+    
+    fn translate_range_loop(
+        &self, 
+        range: &Range, 
+        meta: &mut TranslateMetadata
+    ) -> FragmentKind {
+        // Optimization: Try to resolve range bounds at compile time
+        if let (Some(from_val), Some(to_val)) = (range.from.get_integer_value(), range.to.get_integer_value()) {
+            return self.translate_static_range_loop(range, meta, from_val, to_val);
+        }
+        
+        // Dynamic range
+        let from = range.from.translate(meta);
+        let from_var = meta.push_ephemeral_variable(VarStmtFragment::new("range_start", Type::Int, from))
+            .with_quotes(false).to_frag();
+
+        let to = range.to.translate(meta);
+        let to_var = meta.push_ephemeral_variable(VarStmtFragment::new("range_end", Type::Int, to))
+            .with_quotes(false).to_frag();
+
+        self.translate_dynamic_range_loop(range, meta, from_var, to_var)
+    }
+
+    fn translate_static_range_loop(
+        &self,
+        range: &Range,
+        meta: &mut TranslateMetadata,
+        from_val: isize,
+        to_val: isize
+    ) -> FragmentKind {
+        if range.neq && from_val == to_val {
+            return FragmentKind::Empty;
+        }
+        
+        let iter_name = RawFragment::from(get_variable_name(&self.iter_name, self.iter_global_id)).to_frag();
+        // Define index iterator variable if used
+        let (index_init, index_update) = match (self.iter_index.as_ref(), self.iter_index_global_id) {
+            (Some(index), Some(global_id)) => {
+                let idx_var = get_variable_name(index, Some(global_id));
+                (raw_fragment!(", {idx_var}=0"), raw_fragment!(", {idx_var}++"))
+            },
+            _ => (FragmentKind::Empty, FragmentKind::Empty)
+        };
+
+        let body = self.block.translate(meta);
+        let (op, step) = if from_val <= to_val {(
```

Comment:
`!range.neq && from_val == to_val` is a special case in a static range loop. Actually we can just execute the block once. Maybe this implement should be done as another task.